src/hotspot/share/ci/ciEnv.cpp

```diff
@@ -1647,6 +1647,12 @@ void ciEnv::find_dynamic_call_sites() {
 void ciEnv::dump_compile_data(outputStream* out) {
   CompileTask* task = this->task();
   if (task) {
+#ifdef COMPILER2
+    if (ReplayReduce && compiler_data() != NULL) {
```

Comment:
Is this feature when you are replaying compilation? If yes, add `ReplayCompiles` check too.

Reply:
It seems most useful with ReplayCompiles, but I don't want to require it.  A developer might want to turn on ReplayReduce without ReplayCompiles, to save a replay step, for example when running generated tests from creduce or javafuzzer without an existing replay file.